### PR TITLE
fix nested structs encoding

### LINKED ISSUE #118: encoder does not round trip structs containing struct fields
Given a set of values,
```
var raw = map[string][]string{
	"Inner.A": []string{"1"},
	"Inner.B": []string{"2"},
	"C":       []string{"3"},
}
```
we can decode correctly into the following type `Bar`:
```
type Foo struct {
	A int
	B int
}

type Bar struct {
	Inner Foo
	C     int
}
```

However, encoding `Bar` results in a different map:
```
map[string][]string{
	"A":       []string{"1"},
	"B":       []string{"2"},
	"C":       []string{"3"},
}
```

This behaviour is surprising! I should be able to encode & decode a type and have the results match the input.

I can use different tags for encoding to get around this (e.g. `schema:"Inner.A"`) but it gets tedious (implementation details leak circularly).

[Play Link](https://play.golang.org/p/Kz4Mw9GlVpw)

### PR DESCRIPTION
**Pull request by itmeze**
Fixes #118

**Summary of Changes**

1. Encoding of nested structs creates keys with nested, dot separated, path information. Similar to how decoding works

At the moment there is a difference to how encoding/decoding works with nested structs. Encoding seems to ignore 'nested' struct Information and 'flattens' the key. This pull request fixes that behavior.

### CODE REVIEW

File: encoder_test.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func TestFilled(t *testing.T) {
 	valExists(t, "f07", "seven", vals)
 	valExists(t, "f08", "8", vals)
 	valExists(t, "f09", "1.618000", vals)
-	valExists(t, "F12", "12", vals)
```
Comment (elithrar) [inline encoder_test.go:54]: (What drives removal of this test? Want to make sure there isn’t a behavior change for existing users)

Comment (itmeze) [reply encoder_test.go:54]: Key will be renamed from "F12"  to "F11.F12" as F12 is under F11 struct is that test. Decoding in that case would expect "F11.F12" as a key as well. I am afraid that pull request amends how encoding works so unfortunately there will be a behavior change for existing users :(

**Comment by elithrar** [comment]
Gotcha - thanks for confirming.

In that case I can version a minor release bump to make this clearer -
since it’s not type-breaking, but it is breaking in practice.

**Comment by itmeze** [comment]
here you go: https://github.com/gorilla/schema/pull/144

**Pull request closed without merging**